### PR TITLE
rpc: Auto-derive `ValueHint`s + bunch of output improvements

### PR DESCRIPTION
**Pull request by iljakuklic**
* Add a `ValueHint` derive macro
* Move most manual implementations to the macro
* As a result, some value hints are much more precise
* Update generated outputs
* Various improvements to output rendering

### CODE REVIEW

**Comment by iljakuklic** [comment]
Links to updated docs:
* https://github.com/mintlayer/mintlayer-core/blob/derive-value-hint/node-daemon/docs/RPC.md
* https://github.com/mintlayer/mintlayer-core/blob/derive-value-hint/wallet/wallet-rpc-daemon/docs/RPC.md

File: node-daemon/docs/RPC.md [hunk 1]
```diff
@@ -356,41 +327,76 @@ returns the amount of coins owned by that delegation in that pool.
...
 }
 ```
 
 Returns:
 ```
-{
-    "val": number,
-} OR null
+EITHER OF
+     1) { "val": number }
```
Comment (TheQuantumPhysicist) [inline node-daemon/docs/RPC.md:338]: DecimalAmount is a good thing to return in this function. Returning Amount kind of doesn't make sense...

**Review by TheQuantumPhysicist — APPROVED** [review]

File: node-daemon/docs/RPC.md (hunk 1 above)
Comment (TheQuantumPhysicist) [reply node-daemon/docs/RPC.md:338]: Too many instances... this needs its own PR.

**Review by azarovh — APPROVED** [review]
looks good

**Review by OBorce — APPROVED** [review]

**Pull request merged**